packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts

```diff
@@ -0,0 +1,134 @@
+
+const { testkit, sentryTransport } = sentryTestkit();
+
+// https://github.com/graphile/worker/blob/e3176eab42ada8f4f3718192bada776c22946583/__tests__/helpers.ts#L135
+function makeMockJob(taskIdentifier: string): Job {
+  const createdAt = new Date(Date.now() - 12345678);
+  return {
+    id: String(Math.floor(Math.random() * 4294967296)),
+    queue_name: null,
+    task_identifier: taskIdentifier,
+    payload: {},
+    priority: 0,
+    run_at: new Date(Date.now() - Math.random() * 2000),
+    attempts: 0,
+    max_attempts: 25,
+    last_error: null,
+    created_at: createdAt,
+    updated_at: createdAt,
+    locked_at: null,
+    locked_by: null,
+    revision: 0,
+    key: null,
+    flags: null,
+  };
+}
+
+let helpers: Helpers = makeJobHelpers({}, makeMockJob('remove-old-sent-notifications'), {
+  withPgClient: async () => {
+    throw new Error('not implemented');
+  },
+});
+
+class StubEmail {
+  static shouldThrow = false;
+  static lastSent: any = null;
+
+  async send(email: any) {
+    if (StubEmail.shouldThrow) throw new Error('Intentional error in stub email sending');
+    StubEmail.lastSent = email;
+  }
+}
+
+const cardDropRequest: EmailCardDropRequest = {
+  id: '53a6cd14-ce63-47f0-baaa-0392e4f3d64a',
+  ownerAddress: '0xnotClaimedAddress',
+  emailHash: 'unclaimedhash',
+  verificationCode: 'unclaimedverificationcode',
+  requestedAt: new Date(),
+};
+
+describe('SendEmailCardDropVerificationTask', function () {
+  this.beforeAll(function () {
+    registry(this).register('email', StubEmail);
+  });
+
+  let { getContainer } = setupHub(this);
+  this.beforeEach(async function () {
+    StubEmail.shouldThrow = false;
+    StubEmail.lastSent = null;
+    Sentry.init({
+      dsn: 'https://acacaeaccacacacabcaacdacdacadaca@sentry.io/000001',
+      release: 'test',
+      tracesSampleRate: 1,
+      transport: sentryTransport,
+    });
+    let cardDropQueries = (await getContainer().lookup('email-card-drop-requests', {
+      type: 'query',
+    })) as EmailCardDropRequestsQueries;
+    await cardDropQueries.insert(cardDropRequest);
+  });
+
+  it('sends an email that contains the verification link', async function () {
+    let task = (await getContainer().lookup('send-email-card-drop-verification')) as SendEmailCardDropVerification;
+    let params = new URLSearchParams();
+    params.append('eoa', cardDropRequest.ownerAddress);
+    params.append('verification-code', cardDropRequest.verificationCode);
+    params.append('email-hash', cardDropRequest.emailHash);
+    let link = 'https://card-drop-email.test/email-card-drop/verify' + '?' + params.toString();
+
+    await task.perform({ email: 'anyone@test.test', id: cardDropRequest.id }, helpers);
+
+    expect(StubEmail.lastSent.to).to.equal('anyone@test.test');
+    expect(StubEmail.lastSent.from).to.equal(config.get('aws.ses.supportEmail'));
+    expect(StubEmail.lastSent.text).to.contain(link);
+    expect(StubEmail.lastSent.html).to.contain(link);
```

Comment:
I'm waiting for the template we're meant to use to be finalised before asserting on that, so leaving this a bit relaxed for now.